cf/commands/domain/domains_test.go

```diff
@@ -1,137 +1,379 @@
+						Expect(ui.Outputs).To(ContainSubstrings(
+							[]string{"Routing API URI missing."},
+						))
+					})
+
+				})
+
+				Context("when routing api endpoint is set", func() {
+					Context("when router group does not exist", func() {
+						BeforeEach(func() {
+							domainFields = []models.DomainFields{
+								models.DomainFields{Shared: true,
+									Name:            "Shared-domain1",
+									RouterGroupGuid: "router-group-guid"},
+							}
+							domainRepo.ListDomainsForOrgStub = callBackFunc
+							cmd.Requirements(factory, flagContext)
+							Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+						})
+
+						It("prints the invalid router group message", func() {
+							Expect(ui.Outputs).To(ContainSubstrings(
+								[]string{"FAILED"},
+								[]string{"Invalid router group guid: router-group-guid"},
+							))
+						})
+
+						It("does not print table header", func() {
+							Expect(ui.Outputs).ToNot(ContainSubstrings(
+								[]string{"name", "status", "type"},
+							))
+						})
+					})
+
+					Context("when list router groups returns error", func() {
+						BeforeEach(func() {
+							routingApiRepo.ListRouterGroupsReturns(errors.New("router-group-err"))
+							domainFields = []models.DomainFields{
+								models.DomainFields{Shared: true,
+									Name:            "Shared-domain1",
+									RouterGroupGuid: "my-router-guid1"},
+							}
+							domainRepo.ListDomainsForOrgStub = callBackFunc
+							cmd.Requirements(factory, flagContext)
+							Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+						})
+
+						It("fails with error message", func() {
+							Expect(ui.Outputs).To(ContainSubstrings(
+								[]string{"FAILED"},
+								[]string{"Failed fetching router groups."},
+								[]string{"router-group-err"},
+							))
+						})
+
+					})
+
+					Context("when router group exists", func() {
+						BeforeEach(func() {
+							fakeGroups := models.RouterGroups{
+								models.RouterGroup{
+									Guid: "my-router-guid1",
+									Name: "my-router-name1",
+									Type: "tcp",
+								},
+							}
+							routingApiRepo.ListRouterGroupsStub = func(cb func(models.RouterGroup) bool) error {
+								for _, routerGroup := range fakeGroups {
+									if !cb(routerGroup) {
+										break
+									}
+								}
+								return nil
+							}
+
+							domainFields = []models.DomainFields{
+								models.DomainFields{Shared: true,
+									Name:            "Shared-domain1",
+									RouterGroupGuid: "my-router-guid1"},
+							}
+							domainRepo.ListDomainsForOrgStub = callBackFunc
+							cmd.Requirements(factory, flagContext)
+							cmd.Execute(flagContext)
+						})
+
+						It("prints domain information with router group type", func() {
+							Expect(ui.Outputs).To(ContainSubstrings(
+								[]string{"name", "status", "type"},
+							))
+
+							Expect(ui.Outputs).To(ContainSubstrings(
+								[]string{"Shared-domain1", "shared", "tcp"},
```

Comment:
Line 370 should be moved up to below line 366. This doesn't have to be two expectations.